File: Native/Android/LunarConsole/lunarConsole/src/test/java/spacemadness/com/lunarconsole/dependency/ProviderTest.java
package spacemadness.com.lunarconsole.dependency;

import junit.framework.TestCase;

public class ProviderTest extends TestCase {
	@Override
	protected void setUp() throws Exception {
		super.setUp();
		Provider.clear();
	}

	public void testProvider() {
		MyProviderImpl providable = new MyProviderImpl();

		Provider.register(MyProvider.class, providable);
		assertSame(providable, Provider.of(MyProvider.class));
	}

	public void testMissingProvider() {
		try {
			Provider.of(MyProvider.class);
			fail("Should throw MissingProvidableException");
		} catch (IllegalArgumentException ignored) {
		}
	}

	private interface MyProvider extends ProviderDependency {
	}

	private static class MyProviderImpl implements MyProvider {
	}
}

File: Native/Android/LunarConsole/lunarConsole/src/test/java/spacemadness/com/lunarconsole/dependency/ProviderTest.kt
package spacemadness.com.lunarconsole.dependency

import junit.framework.TestCase

class ProviderTest : TestCase() {
    @Throws(Exception::class)
    override fun setUp() {
        super.setUp()
        Provider.clear()
    }

    fun testProvider() {
        val providable = MyProviderImpl()

        Provider.register(MyProvider::class.java, providable)
        TestCase.assertSame(providable, Provider.of(MyProvider::class.java))
    }

    fun testMissingProvider() {
        try {
            Provider.of(MyProvider::class.java)
            TestCase.fail("Should throw MissingProvidableException")
        } catch (ignored: IllegalArgumentException) {
        }

    }

    private interface MyProvider : ProviderDependency

    private class MyProviderImpl : MyProvider
}